src/internal/pachd/prometheus.go

```diff
@@ -0,0 +1,44 @@
+package pachd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+type prometheusServer struct {
+	port uint16
+}
+
+// listenAndServe listens until ctx is cancelled; it then gracefully shuts down
+// the server, returning once all requests have been handled.
+func (ps prometheusServer) listenAndServe(ctx context.Context) error {
+	var (
```

Comment:
Style question: any reason you chose to use a `var()` block instead of using the `:=` operator on these variables?

Reply:
I prefer `var`: it is visually a separate section which emphasises that variables are being assigned.